core/unit_test/view/TestBasicView.hpp

```diff
@@ -0,0 +1,249 @@
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+#include <type_traits>
+
+using ExecutionSpace = TEST_EXECSPACE;
+
+namespace {
+template <class ExecutionSpace, class Extents>
+auto make_spanning_mdrange_policy_from_extents_impl(const Extents &extents,
+                                                    std::index_sequence<0>) {
+  return Kokkos::RangePolicy<ExecutionSpace>{0, extents.extent(0)};
+}
+
+template <class ExecutionSpace, class Extents, std::size_t... Indices>
+auto make_spanning_mdrange_policy_from_extents_impl(
+    const Extents &extents, std::index_sequence<Indices...>) {
+  using index_type    = typename Extents::index_type;
+  constexpr auto rank = Extents::rank();
+  return Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<rank>>{
+      {(static_cast<index_type>(Indices * 0))...},
+      {extents.extent(Indices)...}};
+}
+
+template <class ExecutionSpace, class Extents>
+auto make_spanning_mdrange_policy_from_extents(const Extents &extents) {
+  return make_spanning_mdrange_policy_from_extents_impl<ExecutionSpace>(
+      extents, std::make_index_sequence<Extents::rank()>{});
+}
+
+template <class T, class ExtentsType>
+static void test_default_constructor() {
+  using extents_type  = ExtentsType;
+  using layout_type   = Kokkos::Experimental::layout_right_padded<>;
+  using accessor_type = Kokkos::Impl::CheckedReferenceCountedAccessor<
+      T, typename ExecutionSpace::memory_space>;
+  using view_type =
+      Kokkos::BasicView<T, extents_type, layout_type, accessor_type>;
+  view_type view;
+
+  EXPECT_FALSE(view.data_handle().has_record());
+  EXPECT_EQ(view.data_handle().get(), nullptr);
+  EXPECT_EQ(view.extents(), extents_type{});
+  EXPECT_EQ(view.data_handle().use_count(), 0);
+  EXPECT_TRUE(view.is_exhaustive());
+  EXPECT_EQ(view.data_handle().get_label(), "");
+  EXPECT_TRUE(view.empty());
+  EXPECT_EQ(view.size(), 0u);
+}
+
+template <class T, class ExtentsType>
+static void test_extents_constructor(const ExtentsType &extents) {
+  using extents_type  = ExtentsType;
+  using layout_type   = Kokkos::Experimental::layout_right_padded<>;
+  using accessor_type = Kokkos::Impl::CheckedReferenceCountedAccessor<
+      T, typename ExecutionSpace::memory_space>;
+  using view_type =
+      Kokkos::BasicView<T, extents_type, layout_type, accessor_type>;
+
+  view_type view("test_view", extents);
+
+  EXPECT_TRUE(view.data_handle().has_record());
+  EXPECT_NE(view.data_handle().get(), nullptr);
+  EXPECT_EQ(view.extents(), extents);
+  EXPECT_EQ(view.data_handle().use_count(), 1);
+  EXPECT_TRUE(view.is_exhaustive());
+  EXPECT_EQ(view.data_handle().get_label(), "test_view");
+  size_t expected_size = 1;
+  // Avoid pointless comparison of unsigned warning for rank==0
+  for (int r = 0; r < static_cast<int>(view_type::rank()); r++)
+    expected_size *= extents.extent(r);
+  EXPECT_EQ(view.size(), expected_size);
+  EXPECT_EQ(view.empty(), expected_size == 0u);
```

Comment:
`0u` is not "covered".
I would rather assert that the size is greater than 0 and that it is not empty.

Reply:
covering 0